src/flow/message.ts

```diff
@@ -87,15 +87,17 @@ export function printMessage(fileName: string, exportMap: ExportMap, messageDesc
         if (valueType === BYTES_TYPE) {
           valueTypeName = "Uint8Array | string";
         }
-        if (valueType === MESSAGE_TYPE) {
-          valueTypeName += `$${objectTypeName}`;
-        }
         if (valueType === ENUM_TYPE) {
           valueTypeName = `$Values<typeof ${valueTypeName}>`;
         }
         printer.printIndentedLn(`get${withUppercase}Map: () => jspb.Map<${keyTypeName}, ${valueTypeName}>;`);
         printer.printIndentedLn(`clear${withUppercase}Map: () => void;`);
-        toObjectType.printIndentedLn(`${camelCaseName}Map: Array<[${keyTypeName}${keyType === MESSAGE_TYPE ? `$${objectTypeName}` : ""}, ${valueTypeName}]>,`);
+
+        let valueTypeNameToObject = valueTypeName
+        if (valueType === MESSAGE_TYPE) {
+          valueTypeNameToObject += `$${objectTypeName}`;
+        }
+        toObjectType.printIndentedLn(`${camelCaseName}Map: Array<[${keyTypeName}${keyType === MESSAGE_TYPE ? `$${objectTypeName}` : ""}, ${valueTypeNameToObject}]>,`);
```

Comment:
I might miss somewhere in the code, but I don't quite understand why it matters here.
This seems to generate the string with  `Map: Array<>`, but the changed test case doesn't have the `Array` in the generated string.

Reply:
That's the expected behavior since `toObjectType` is the JSON type (postfixed with `$AsObject`) code printer, which should continue to have the postfix. Adding the `valueTypeNameToObject` variable preserves that behavior as before, but changes the value of `valueTypeName` on line 93 to not include the JSON postfix (since it isn't JSON).

Reply:
I see. Thanks for explaining. Line 93 is what matters here.
Then Why do we need to redeclare `valueTypeNameToObject` then? We can simply move previous line 90-92 down to right before the line of `toObjectType.printIndentedLn`

Reply:
Oh, yeah looks like I can do that. I didn't realize `valueTypeName` could be safely changed.